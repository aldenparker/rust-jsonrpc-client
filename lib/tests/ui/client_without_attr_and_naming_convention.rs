use jsonrpc_client::{Response, SendRequest, Url};
use serde::de::DeserializeOwned;

#[jsonrpc_client::api]
pub trait Math {
    async fn subtract(&self, subtrahend: i64, minuend: i64) -> i64;
}

struct InnerClient;

#[async_trait::async_trait]
impl SendRequest for InnerClient {
    type Error = std::io::Error;

    async fn send_request<P>(&self, endpoint: Url, body: String) -> Result<Response<P>, Self::Error>
    where
        P: DeserializeOwned,
    {
        unimplemented!()
    }
}

#[jsonrpc_client::implement(Math)]
pub struct Client {
    inner_client: InnerClient,
}

fn main() {}
