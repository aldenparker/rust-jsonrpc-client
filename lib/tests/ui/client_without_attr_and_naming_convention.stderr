error: struct needs to have either a field named `inner` or one tagged with `#[jsonrpc_client(inner)]`
  --> $DIR/client_without_attr_and_naming_convention.rs:24:19
   |
24 |   pub struct Client {
   |  ___________________^
25 | |     inner_client: InnerClient,
26 | | }
   | |_^
